mopidy_mpd/protocol/album_art.py

```diff
@@ -0,0 +1,108 @@
+from mopidy_mpd import protocol
+from urllib.request import urlopen
+
+
+def _get_art_url(self, track):
+    images = self.core.library.get_images([track.uri]).get()
+    if images[track.uri]:
+        largest_image = sorted(
+            images[track.uri], key=lambda i: i.width or 0, reverse=True
+        )[0]
+        return largest_image.uri
+
+
+cover_cache = {}
+
+
+@protocol.commands.add("albumart")
+def albumart(context, uri, offset):
+    """
+    *musicpd.org, the music database section:*
+
+        ``albumart {URI} {OFFSET}``
+
+        Locate album art for the given song and return a chunk
+        of an album art image file at offset OFFSET.
+
+        This is currently implemented by searching the directory
+        the file resides in for a file called cover.png, cover.jpg,
+        cover.tiff or cover.bmp.
+
+        Returns the file size and actual number of bytes read at
+        the requested offset, followed by the chunk requested as
+        raw bytes (see Binary Responses), then a newline and the completion code.
+
+        Example::
+
+            albumart foo/bar.ogg 0
+            size: 1024768
+            binary: 8192
+            <8192 bytes>
+            OK
+
+    .. versionadded:: 0.21
+        New in MPD protocol version 0.21
+    """
+    global cover_cache
+
+    if uri not in cover_cache:
+        (tlid, track) = context.core.playback.get_current_tl_track().get()
```

Comment:
See my initial comments from Dec 20 (https://github.com/mopidy/mopidy-mpd/pull/57#issuecomment-1358686004). Different MPD clients try to pre-fetch all the album images for all the albums in the local library. The current code will put the same album image of the track that's currently playin on all the albums in the library.
I tested this with different MPD client on all platforms (Android, Windows, Linux, Mac), that's why my code turned out to be more complex then you might expect.

Reply:
Indeed, that's my bad, in fact I'm uncertain why that would be required in the first place. I changed it to simply use `uri` in [9fc87dbc](https://github.com/mopidy/mopidy-mpd/compare/a0e10f935a9fc6b147d588df1a7536a47419548d..daad0598cc2b6f2bbcac750e038fed185cd5273f), thanks for the notice!

Reply:
> The fundamental problem here is relying on the internal behaviour of Mopidy-Local, an entirely separate package which is free to change at any time. If you want to interface with a backend, you must do so through the backend-agnostic API provided by Mopidys Core.

OK, clear.

>I've gone back and forth over this over the months (years?!) but I think it's fine in the case of Mopidy-Local since that's a restriction it already has today. And once you support getting the image data over HTTP, you've then also supported most streaming service Backends.

Personally I don't like the dependency on HTTP, I don't use it and it's a strange detour to access files on the same system,  but if it solves to problem then I can accept that. Note that my code already supports HTTP calls, but of course has a lot of extra code to support file level access to the images.

>We can work that way if necessary. In which case we'll move this work into a feature branch rather than merge into master to avoid releasing a half-broken implementation.

OK, it's only a bit unfortunate that I receive this feedback last minute without getting a change to fix things. But if we agree on the HTTP approach and you want help I can create a new branch when I find the time for it.

Reply:
@gvc0 In that case feel free to use [leso-kn:feature/album-art](https://github.com/leso-kn/mopidy-mpd/tree/feature/album-art-mopidy-local) as a baseline, it contains your work plus a broad extension of the `test_albumart.py` unit-test to cover most of the new code :)

I'm marking this discussion as resolved as the relevant code sections have been adjusted.

Reply:
OK, thx